fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/registry/FabricBrewingRecipeRegistry.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.registry;
+
+import java.util.Objects;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.PotionItem;
+import net.minecraft.potion.Potion;
+import net.minecraft.recipe.BrewingRecipeRegistry;
+import net.minecraft.recipe.Ingredient;
+
+/**
+ * Counterpart of {@link BrewingRecipeRegistry} with methods that allow adding recipes which use Ingredients instead of Items.
+ */
+public final class FabricBrewingRecipeRegistry {
+	private FabricBrewingRecipeRegistry() {
+	}
+
+	/**
+	 * Register a recipe for brewing one potion type into another (e.g. regular to splash).
+	 * Only one recipe is necessary for all potions of the input type to be brewable into the output type using the ingredient.
+	 * Use {@link BrewingRecipeRegistry#registerPotionType(Item)} to register new potion types.
+	 * @param input the input potion type (e.g. regular potion)
+	 * @param ingredient the required ingredient (e.g. gunpowder)
+	 * @param output the output type  (e.g. splash potion)
+	 * @see BrewingRecipeRegistry#registerItemRecipe(Item, Item, Item)
+	 */
+	public static void registerItemRecipe(PotionItem input, Ingredient ingredient, PotionItem output) {
```

Comment:
The vanilla method takes an 'Item' and not a 'PotionItem' as a pram and then validates that it is a `PotionItem` I think it may be best to mimick this behaviour?

All of the fields in vanilla's `Items` class as an `Item` so you would need to cast it to use it in this method reducing/removing the benefit of using the more specific type.

`BrewingRecipeRegistry.Recipe` seems to allow for any `Item` type.

Reply:
Requiring a cast for vanilla potions isn't clean, but neither is a method that declares a less specific type but actually requires a more specific type. My original thought was it's better to detect an incorrect type at compile time or with IDE error checking, rather than at runtime.

> `BrewingRecipeRegistry.Recipe` seems to allow for any `Item` type.

If you're proposing allowing for brewing any item into any item, that looks like it would only require small alterations to `net.minecraft.block.entity.BrewingStandBlockEntity#isValid` & `net.minecraft.screen.BrewingStandScreenHandler.PotionSlot#matches`.

Reply:
modmuss was mentioning that the natural extension of the vanilla methods would be to do the instanceof check at the beginning of the method

Reply:
> neither is a method that declares a less specific type but actually requires a more specific type

I agree that both options are suboptimal, but here it's obvious from looking at the method's body that you need a potion item (could also be said explicitly in the javadoc). In doubt we generally mimic mojang's decisions even if they're not optimal.

Reply:
Either is fine for me as well, your choice Daomephsta.

Reply:
Lets stick with what is here 👍